### PR TITLE
feat(CFW): add ips_switch_status、ips_protect_mode attribute to cfw_firewall resource

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
add ips_switch_status、ips_protect_mode attribute to cfw_firewall resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.  add ips_switch_status、ips_protect_mode attribute to cfw_firewall resource
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cfw" TESTARGS="-run TestAccFirewall_ips"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cfw -v -run TestAccFirewall_ips -timeout 360m -parallel 4
=== RUN   TestAccFirewall_ips
=== PAUSE TestAccFirewall_ips
=== CONT  TestAccFirewall_ips
--- PASS: TestAccFirewall_ips (482.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cfw       483.026s
```

### CODE REVIEW

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go [hunk 1]
```diff
@@ -650,50 +722,146 @@ func resourceFirewallUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildUpdateIpsProtectModeBodyParams(mode, objectID)),
+	}
+
+	_, err = updateProtectModeClient.Request("POST", updateProtectModePath, &updateProtectModeOpt)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (houpeng80) [inline huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go]: ditto

File: docs/resources/cfw_firewall.md [hunk 2]
```diff
@@ -68,6 +68,30 @@ resource "huaweicloud_cfw_firewall" "test" {
...
+    key = "value"
+    foo = "bar"
+  }
+
+  charging_mode = "prePaid"
+  period_unit   = "month"
+  period        = 1
+  auto_renew    = false
+  ips_switch_status = 1
+  ips_protection_mode  = 1
```
Comment (houpeng80) [inline docs/resources/cfw_firewall.md]:
```suggestion
    ips_switch_status = 1
  ips_protection_mode  = 1
  
  charging_mode = "prePaid"
  period_unit   = "month"
  period        = 1
  auto_renew    = false
```
align the equal signs, and fix other places

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go [hunk 3]
```diff
@@ -650,50 +722,146 @@ func resourceFirewallUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildUpdateIpsSwitchStatusBodyParams(status, objectID)),
+	}
+
+	_, err = updateSwitchStatusClient.Request("POST", updateSwitchStatusPath, &updateSwitchStatusOpt)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (houpeng80) [inline huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go]:
```suggestion
	return err
```

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go (hunk 1 above)
Comment (profoundwu) [reply huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go]: fixed

File: docs/resources/cfw_firewall.md (hunk 2 above)
Comment (profoundwu) [reply docs/resources/cfw_firewall.md]: fixed

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go (hunk 3 above)
Comment (profoundwu) [reply huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go]: fixed

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go [hunk 4]
```diff
@@ -607,11 +633,57 @@ func flattenGetFirewallResponseBodyProtectObject(resp interface{}) []interface{}
...
+
+	return jmespath.Search("data.virtual_patches_status", getIpsSwitchStatusRespBody)
+}
+
 func resourceFirewallUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	cfg := meta.(*config.Config)
 	region := cfg.GetRegion(d)
 
-	var objectID string
+	var objectID, internetBorderObjectID string
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go]:
```suggestion
	var vpcBoderObjectID, internetBorderObjectID string
```

File: huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go [hunk 5]
```diff
@@ -650,50 +722,138 @@ func resourceFirewallUpdate(ctx context.Context, d *schema.ResourceData, meta in
...
+			}
 		}
 	}
 
-	if objectID != "" {
-		var (
-			updateEastWestFirewallHttpUrl = "v1/{project_id}/firewall/east-west/protect"
-			updateEastWestFirewallProduct = "cfw"
-		)
+	if d.IsNewResource() || d.HasChange("east_west_firewall_status") {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/cfw/resource_huaweicloud_cfw_firewall.go:739]:
```suggestion
	d.HasChange("east_west_firewall_status") {
```

**Pull request merged**